src/core/utils/filternodeutils.js

```diff
@@ -45,3 +45,35 @@ export function pruneFilterNodes (filterNodes, hiddenFields) {
       return !hiddenFields.includes(fieldId);
     });
 }
+
+/**
+ * Check whether two non-combined (i.e. do not contain an array of child Filters) filters
+ * are equal to each other.
+ * @param {Filter} filterNode1
+ * @param {Filter} filterNode2
+ * @returns {boolean}
+ */
+export function haveEqualSimpleFilters (filterNode1, filterNode2) {
```

Comment:
Does this method determine if two SimpleFilterNodes are equal? If so, I think this is better suited as an "equal" method for the SimpleFilterNode class. The "equal" method takes in another SimpleFilterNode and compares it to the current one.

Reply:
it does not, it only compares their filters

Reply:
actually, it can probably be turned into that, though. the filternodes created from the nlp filters also have a valid display value